### ISSUE TITLE
Datepicker calendar not rendered right when used as ag-grid cell editor

### ISSUE THREAD

**Issue by nicolae536** [angular/material2#6656]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?

The calendar should be rendered right when it's open.
#### What is the current behavior?

The calendar is rendered on the top left corner of the window
#### What are the steps to reproduce?
When having an md-input connected to the datepicker and the mat-input-inderline has display: none; the calendar is opened on the top left corner of the page.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U
![image](https://user-images.githubusercontent.com/9453094/29752768-bea5b6e0-8b6c-11e7-9dfc-6c686aaaa782.png)

#### What is the use-case or motivation for changing an existing behavior?
The datepicker should render under the connected input

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular - 4.3.1
Material design - 2.0.0-beta.8
CDK - 2.0.0-beta.8

#### Is there anything else we should know?

I found this issue because my css for ag-grid hides the mat-input-underline.

```
 .mat-input-underline {
     display: none;
 }
```

**Comment by donroyco** [comment]
Please provide a Plunker to reproduce the mentioned issue.

**Comment by nicolae536** [comment]
Do I still need to provide a plunker for it ? I found the real problem and changed the description, is this a bug or is supposed to not work if the mat-input-underline is not visible ?

**Comment by willshowell** [comment]
@nicolae536 I can't speak to whether it is/isn't a valid bug, but currently the datepicker _does_ require the underline for position attachment.

https://github.com/angular/material2/blob/ec4ea060a519282b362db0dd066aab54c4994cc1/src/lib/datepicker/datepicker-input.ts#L251-L257

https://plnkr.co/edit/LJFXEinH661F70aLPaLi?p=preview

**Comment by mmalerba** [comment]
This is expected if you put `display:none` on the underline, since the datepicker attaches itself at the underline position. You could try `height:0; visibility:hidden; pointer-events:none;` to visually hide it, but leave it in the layout so the datepicker knows where to place the popup.

**Comment by vesrah** [comment]
I came across this bug report through some creative Googling.  I also had a similar issue, but it was just mis-alignment for me, rather than completely sitting at the top left of the screen.  My CSS fix could be summed up as:
Added class to containing mat-form-field, for example we will call it mat-datepicker, then the following pseudo CSS.
`.mat-datepicker { position: relative; }`
`.mat-datepicker .mat-input-underline { position: absolute; z-index: -1; opacity: 0; bottom: 0 }`

Again, the classes/rules here are just to illustrate.
![screen shot 2018-01-08 at 11 00 03 am](https://user-images.githubusercontent.com/9937308/34685014-3a67dde8-f464-11e7-94f9-17d7002f4247.png)
